client/my-sites/current-site/index.jsx

```diff
@@ -58,13 +57,11 @@ const CurrentSite = React.createClass( {
 		};
 	},
 
-	componentWillUpdate() {
-		const { selectedSite } = this.props;
-
-		if ( selectedSite && this.prevSelectedSite !== selectedSite && ! selectedSite.jetpack ) {
-			UpgradesActions.fetchDomains( selectedSite.ID );
+	componentDidUpdate( prevProps ) {
```

Comment:
This also matches the last React recommendations https://twitter.com/acdlite/status/818930692109242369 👍